MAUI/DataForm/Validation.md

```diff
@@ -0,0 +1,276 @@
+
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.Validate(new List<string>() {"FirstName", "Age" });
+
+{% endhighlight %}
+{% endtabs %}
+
+You can determine whether the data form or property is valid or not by using the `Validate` method.
+
+{% tabs %}
+{% highlight C# %}
+
+        bool isValid = this.dataForm.Validate();
+
+        List<string> propertyNames = new List<string>();
+        propertyNames.Add("FirstName");
+        bool isPropertyValid = this.dataForm.Validate(propertyNames);
+
+{% endhighlight %}
+{% endtabs %}
+
+If the data form or property is valid, `true` will be returned. Or else `false` will be returned.
+
+## Valid message
+
+If the values are correct, you can show the [ValidMessage] as like error message, the valid message will also be displayed at the bottom of the editor.
+
+{% tabs %}
+{% highlight C# %}
+
+    [DataFormDisplayOptions(ValidMessage = "Password strength is good")]
+    [Required(ErrorMessage = "Please enter the password")]
+    [RegularExpression(@"^(?=.*[a-z])(?=.*[A-Z])[a-zA-Z\d]{8,}$", ErrorMessage = "A minimum 8-character password should contain a combination of uppercase and lowercase letters.")]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Validate the data form
+
+You can get the validation details of the all the editors of data form using the [ValidateForm] event of the data form.
+
+Note: This event will be raised once after the manual validation call using [SfDataForm.Validate()] method.
+
+{% endhighlight %}
+{% highlight C# %}
+
+this.dataForm.ValidateForm += this.OnDataFormValidateForm;
+
+    private void OnDataFormValidateForm(object sender, DataFormValidateFormEventArgs e)
+    {
+        object dataObject = e.DataObject;
+        var values = e.NewValues;
+        var errorMessage = e.ErrorMessage;
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Validate the specific editor
+
+[ValidateProperty] event allows you to validate specific editor in the data form. You can set `Isvalid`, `ErrorMessage` and `ValidMessage` of [DataFormValidatePropertyEventArgs].
+
+{% endhighlight %}
+{% highlight C# %}
+
+this.dataForm.ValidateProperty += this.OnDataFormValidateProperty;
+
+    private void OnDataFormValidateProperty(object sender, DataFormValidatePropertyEventArgs e)
+    {
+        bool isValid = e.IsValid;
+        string propertyName = e.PropertyName;
+        object newValue = e.NewValue;
+        object currentValue = e.CurrentValue;
+        string errorMessage = e.ErrorMessage;
+        string validMessage = e.ValidMessage;
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Validation label appearance customization
+
+The data form supports to customize the style of both error and valid message label style easily.
+
+#### Customize error label text style
+
+The `error label style` can be customized by changing the [ErrorLabelTextStyle] property of [SfDataForm].
```

Comment:
The `error label style` can be customized by changing the [ErrorLabelTextStyle] property of the [SfDataForm].

Reply:
content changed